test/e2e-go/debian/Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM debian:stable
+RUN \
+apt-get update && \
+apt-get update && \
+apt-get install -y apt-transport-https && \
+apt-get update && \
+apt-get install -y gnupg2 curl software-properties-common && \
+apt-key adv --fetch-keys https://releases.algorand.com/key.pub && \
+add-apt-repository "deb https://releases.algorand.com/deb/ stable main" && \
+apt-get update && \
+apt-get install -y apt-transport-https && \
+apt-get update && \
+apt-get install -y algorand && \
```

Comment:
I'm surprised by the redundancy in this script, I've never seen any process need that. Just one `apt-get update` to start and one after adding the repo.
Also, it looks like this only tests an already deployed .deb package, but we'd want to test _before_ deploying.
See current tests on the deb as we build it:
https://github.com/algorand/go-algorand/blob/master/scripts/build_release.sh#L119-L155

Reply:
Removed some of the updates from docker file.

Reply:
it still has the problem of only testing debian packages we have already deployed to the public release server. it should be able to test a local .deb file.
Also, Docker will cache builds and build steps, so putting everything but the `install algorand` in the Dockerfile should make it faster by caching all the prep work and then running the test on top of that.

Reply:
The script I linked above sets up a local .deb repo and http server to make `apt-get install algorand` work from a just-built .deb package being tested.

Reply:
Updated so that the test is invoked as part of the build_release.sh script.